library/entropy.c

```diff
@@ -285,87 +285,88 @@ int mbedtls_entropy_gather( mbedtls_entropy_context *ctx )
 
 int mbedtls_entropy_func( void *data, unsigned char *output, size_t len )
 {
-    int ret, count = 0, i, done;
+    int ret, count = 0, j = 0, i, done;
     mbedtls_entropy_context *ctx = (mbedtls_entropy_context *) data;
     unsigned char buf[MBEDTLS_ENTROPY_BLOCK_SIZE];
 
-    if( len > MBEDTLS_ENTROPY_BLOCK_SIZE )
-        return( MBEDTLS_ERR_ENTROPY_SOURCE_FAILED );
-
-#if defined(MBEDTLS_ENTROPY_NV_SEED)
-    /* Update the NV entropy seed before generating any entropy for outside
-     * use.
-     */
-    if( ctx->initial_entropy_run == 0 )
+    while (len > 0)
     {
-        ctx->initial_entropy_run = 1;
-        if( ( ret = mbedtls_entropy_update_nv_seed( ctx ) ) != 0 )
-            return( ret );
-    }
+#if defined(MBEDTLS_ENTROPY_NV_SEED)
+        /* Update the NV entropy seed before generating any entropy for outside
+         * use.
+         */
+        if( ctx->initial_entropy_run == 0 )
+        {
+            ctx->initial_entropy_run = 1;
+            if( ( ret = mbedtls_entropy_update_nv_seed( ctx ) ) != 0 )
```

Comment:
I am not sure about this, do we really want to update the nv_seed for very large polls?

Reply:
This API was to define an NV seed according to the desired size of the seed. I think it's valid